code/datums/components/footstep.dm

```diff
@@ -37,67 +60,48 @@
 	if(steps % 2)
 		return
 
-	if(!LM.has_gravity(T) && steps != 0) // don't need to step as often when you hop around
+	if(steps != 0 && !LM.has_gravity(T)) // don't need to step as often when you hop around
 		return
+	return T
 
-	//begin playsound shenanigans//
-
-	//for barefooted non-clawed mobs like monkeys
-	if(isbarefoot(LM))
-		playsound(T, pick(GLOB.barefootstep[T.barefootstep][1]),
-			GLOB.barefootstep[T.barefootstep][2] * v,
-			TRUE,
-			GLOB.barefootstep[T.barefootstep][3] + e)
+/datum/component/footstep/proc/play_simplestep()
+	var/turf/open/T = prepare_step()
+	if(!T)
 		return
-
-	//for xenomorphs, dogs, and other clawed mobs
-	if(isclawfoot(LM))
-		if(isalienadult(LM)) //xenos are stealthy and get quieter footsteps
-			v /= 2
-			e -= 5
-
-		playsound(T, pick(GLOB.clawfootstep[T.clawfootstep][1]),
-				GLOB.clawfootstep[T.clawfootstep][2] * v,
-				TRUE,
-				GLOB.clawfootstep[T.clawfootstep][3] + e)
+	if(isfile(footstep_sounds))
+		playsound(T, footstep_sounds, volume)
 		return
-
-	//for megafauna and other large and imtimidating mobs such as the bloodminer
-	if(isheavyfoot(LM))
-		playsound(T, pick(GLOB.heavyfootstep[T.heavyfootstep][1]),
-				GLOB.heavyfootstep[T.heavyfootstep][2] * v,
-				TRUE,
-				GLOB.heavyfootstep[T.heavyfootstep][3] + e)
+	var/turf_footstep
+	switch(footstep_type)
+		if(FOOTSTEP_MOB_CLAW)
+			turf_footstep = T.clawfootstep
+		if(FOOTSTEP_MOB_BAREFOOT)
+			turf_footstep = T.barefootstep
+		if(FOOTSTEP_MOB_HEAVY)
+			turf_footstep = T.heavyfootstep
+		if(FOOTSTEP_MOB_SHOE)
+			turf_footstep = T.footstep
+	if(!turf_footstep)
 		return
```

Comment:
you can use else in a switch

Reply:
else doesn't help here. Just because footstep_type exists doesn't mean that turf_footstep exists.

Reply:
ah, yeah if T.*_footstep can be null then that wont help